src/main/java/org/apache/maven/plugins/assembly/archive/DefaultAssemblyArchiver.java

```diff
@@ -293,10 +294,7 @@ protected Archiver createArchiver(
         final List<FileSelector> extraSelectors = new ArrayList<>();
         final List<ArchiveFinalizer> extraFinalizers = new ArrayList<>();
         if (archiver instanceof JarArchiver) {
-            if (mergeManifestMode != null) {
-                ((JarArchiver) archiver)
-                        .setFilesetmanifest(JarArchiver.FilesetManifestConfig.valueOf(mergeManifestMode));
-            }
+            configureJarArchiver((JarArchiver) archiver, mergeManifestMode, configSource.getJarArchiveConfiguration());
```

Comment:
I would expect that this applies to all JAR derived formats? WAR, EAR, CAR, etc?

Reply:
yes - currently ear, rar, war 
https://codehaus-plexus.github.io/plexus-archiver/apidocs/org/codehaus/plexus/archiver/jar/JarArchiver.html

I would like to extends another types, eg for car we only have a UnArchiver
https://codehaus-plexus.github.io/plexus-archiver/apidocs/org/codehaus/plexus/archiver/car/package-summary.html